test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/scaling/KieServerHttpScalingIntegrationTest.java

```diff
@@ -0,0 +1,117 @@
+import org.kie.cloud.maven.MavenDeployer;
+import org.kie.cloud.maven.constants.MavenConstants;
+import org.kie.server.api.model.KieContainerResourceList;
+import org.kie.server.api.model.KieContainerStatus;
+import org.kie.server.api.model.KieServerInfo;
+import org.kie.server.api.model.ServiceResponse;
+import org.kie.server.client.KieServicesClient;
+import org.kie.server.controller.api.model.runtime.ServerInstanceKey;
+import org.kie.server.controller.api.model.spec.ContainerSpec;
+import org.kie.server.integrationtests.controller.client.KieServerMgmtControllerClient;
+
+public class KieServerHttpScalingIntegrationTest extends AbstractCloudIntegrationTest<WorkbenchWithKieServerScenario> {
+
+    private KieServerMgmtControllerClient kieControllerClient;
+    private KieServicesClient kieServerClient;
+
+    @Override
+    protected WorkbenchWithKieServerScenario createDeploymentScenario(DeploymentScenarioBuilderFactory deploymentScenarioFactory) {
+        return deploymentScenarioFactory.getWorkbenchWithKieServerScenarioBuilder()
+                .withExternalMavenRepo(MavenConstants.getMavenRepoUrl(), MavenConstants.getMavenRepoUser(), MavenConstants.getMavenRepoPassword())
+                .build();
+    }
+
+    @Before
+    public void setUp() {
+        MavenDeployer.buildAndDeployMavenProject(ClassLoader.class.getResource("/kjars-sources/definition-project-snapshot").getFile());
+
+        kieControllerClient = KieServerControllerClientProvider.getKieServerMgmtControllerClient(deploymentScenario.getWorkbenchDeployment());
+        kieServerClient = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployment());
+    }
+
+    @Test
+    public void testConnectionBetweenDeployables() {
+        String kieServerId = getKieServerId(kieServerClient);
+
+        verifyServerTemplateContainsKieServers(kieServerId, 1);
+
+        deployAndStartContainer();
+
+        verifyContainerIsDeployed(kieServerClient, CONTAINER_ID);
+        verifyServerTemplateContainsContainer(kieServerId, CONTAINER_ID);
+
+        scaleKieServerTo(3);
+
+        verifyServerTemplateContainsKieServers(kieServerId, 3);
+
+        scaleKieServerTo(1);
+
+        verifyServerTemplateContainsKieServers(kieServerId, 1);
+    }
+
+    private String getKieServerId(KieServicesClient kieServerClient) {
+        ServiceResponse<KieServerInfo> serverInfo = kieServerClient.getServerInfo();
+        assertThat(serverInfo.getType()).isEqualTo(ServiceResponse.ResponseType.SUCCESS);
+
+        return serverInfo.getResult().getServerId();
+    }
+
+    private void verifyContainerIsDeployed(KieServicesClient kieServerClient, String containerId) {
+        ServiceResponse<KieContainerResourceList> containers = kieServerClient.listContainers();
+        assertThat(containers.getType()).isEqualTo(ServiceResponse.ResponseType.SUCCESS);
+
+        assertThat(containers.getResult().getContainers()).hasSize(1);
```

Comment:
Just a nitpick: I'd extract `containers.getResult().getContainers()` to variable to make bit more readable.

Reply:
Adjusted